android-core/src/main/java/com/mparticle/internal/database/services/MParticleDBManager.java

```diff
@@ -300,7 +309,17 @@ private void createUploads(Map<BatchId, MessageBatch> uploadMessagesByBatchId, M
                 uploadMessage.setIdentities(identities);
                 JSONObject userAttributes = findUserAttributeState(messages, batchId.getMpid());
                 uploadMessage.setUserAttributes(userAttributes);
-                UploadService.insertUpload(db, uploadMessage, configManager.getApiKey());
+
+                JSONObject batch = uploadMessage;
+                if (options.getBatchCreationListener() != null) {
+                    batch = options.getBatchCreationListener().onBatchCreated(batch);
+                }
```

Comment:
we should wrap this in a try/catch so the client code doesn't blow up the rest of this procedure if their code throws an exception